src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java

```diff
@@ -144,4 +147,25 @@ public String getPagedResponseTypeInnerName(
   public String getFullyQualifiedApiWrapperClassName(Interface service) {
     return getPackageName() + "." + getApiWrapperClassName(service);
   }
+
+  @Override
+  public String getRandomStringValue(String randomString) {
+    String randomValueGen = "System.currentTimeMillis()";
+    String delimiter = ",";
+    String[] splitted =
+        FieldStructureParser.stripQuotes(randomString)
+            .replace(InitFieldConfig.RANDOM_TOKEN, delimiter + randomValueGen + delimiter)
+            .split(delimiter);
+    ArrayList<String> stringParts = new ArrayList<>();
+    for (String token : splitted) {
+      if (token.length() > 0) {
+        if (token.equals(randomValueGen)) {
+          stringParts.add(token);
+        } else {
+          stringParts.add("\"" + token + "\"");
+        }
+      }
+    }
+    return String.join(" + ", stringParts);
```

Comment:
I think I'd be more comfortable with `randomValueGen` being injected later than you do - i.e. in your `replace(...)` call, use RANDOM_TOKEN instead of `randomValueGen`, and then when you are adding to `stringParts`, use `"System.currentTimeMillis()"` there.